### PR TITLE
Add a tuple decoder (along with tests and docs).

### LINKED ISSUE #22: Support for tuples.
It would be nice to have support for tuples: https://github.com/Microsoft/TypeScript/blob/master/doc/spec.md#3.3.3

### PR DESCRIPTION
**Pull request by rhofour**
This fixed #22

### CODE REVIEW

**Comment by joanllenas** [comment]
This looks really good @rhofour, nice job!
I'll review/merge it later today.
Thanks!

**Comment by rhofour** [comment]
I had some time and could use a tuple decoder so I figured this was a good time to learn some of the more advanced bits of TypeScript.

The only thing I wish I could change about it is the eliminate the cast I have to do when returning the value. I believe it's always perfectly type-safe, but I haven't figured out how to get the compiler to see that.

File: src/json-decoder.ts [hunk 1]
```diff
@@ -472,6 +472,44 @@ export namespace JsonDecoder {
...
+    decoderName: string
+  ): Decoder<TupleOfResults<T>> => {
+    return new Decoder<TupleOfResults<T>>(json => {
+      if (json instanceof Array) {
+        const arr = [];
+        if (json.length !== decoders.length) {
+          return err<TupleOfResults<T>>($JsonDecoderErrors.tupleLengthMismatchError(
+            decoderName, json, decoders));
+        }
+        for (let i =-0; i < json.length; i++) {
```
Comment (joanllenas) [inline src/json-decoder.ts:495]: What is this `-0` for?

**Review by joanllenas — COMMENTED** [review]
Other than that the `-0` question, this is good to go. 
Great job, and a nice feature to have!

File: src/json-decoder.ts (hunk 1 above)
Comment (rhofour) [reply src/json-decoder.ts]: Just a consequence of typing on my laptop keyboard rather than my usual setup. I just sent a new commit to fix it.

Comment (joanllenas) [reply src/json-decoder.ts]: Ok, I thought that maybe this was some kind of weird performance optimization technique I was not aware of. This things exists! :)
I'll merge release this afternoon and I'll let you know when I'm done.
Thanks!

**Comment by joanllenas** [comment]
And here it is!
https://github.com/joanllenas/ts.data.json/releases/tag/v1.5.0
Thanks!

**Pull request merged**